Bottom: 1eb9f9c443bdead1f6590c270f2a10f212412b42
Top:    2dfc35b1331b122d84820efec0a6c97c930feb6a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-17 20:01:04 +0000

polygon1.c: Temporarily disable the seg_in_region test

The code does not handle the bounding region of arcs yet


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 4ee97aa..ed4c2bc 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -633,6 +633,7 @@ adjust_tree (rtree_t * tree, struct seg *s)
   return 0;
 }
 
+#if 0  /* TURNS OUT DISABLING THIS ACTUALLY (VERY MARGINALLY) SPEEDS THINGS UP! */
 /*
  * seg_in_region()
  * (C) 2006, harry eaton
@@ -654,6 +655,7 @@ seg_in_region (const BoxType * b, void *cl)
     return 0;
   return 1;			/* might intersect */
 }
+#endif
 
 /* Prepend a deferred node-insersion task to a list */
 static insert_node_task *
@@ -877,7 +879,7 @@ contour_bounds_touch (const BoxType * b, void *cl)
       info.tree = rtree_over->tree;
       if (info.tree)
 	if (UNLIKELY (r_search (info.tree, &info.s->box,
-				seg_in_region, seg_in_seg, &info)))
+				NULL /* seg_in_region */, seg_in_seg, &info)))
 	  assert (0); /* XXX: Memory allocation failure */
     }
   while ((av = NEXT_EDGE (av)) != &looping_over->head);
